src/main/java/com/aws/greengrass/shadowmanager/sync/SyncOnInteractionHandler.java

```diff
@@ -0,0 +1,274 @@
+                .kv(LOG_THING_NAME_KEY, syncConfiguration.getThingName())
+                .kv(LOG_SHADOW_NAME_KEY, syncConfiguration.getShadowName())
+                .log("register thing shadow to sync on interaction");
+
+        if (syncConfigurations.contains(syncConfiguration)) {
+            return;
+        }
+
+        if (thingIsNotAddedOnInteraction(syncConfigurations, syncConfiguration.getThingName())) {
+            logger.atWarn()
+                    .kv(LOG_THING_NAME_KEY, syncConfiguration.getThingName())
+                    .kv(LOG_SHADOW_NAME_KEY, syncConfiguration.getShadowName())
+                    .log("can not register thing shadow on interaction because the thing has other non-interaction "
+                            + "shadows");
+            return;
+        }
+
+        // clone syncConfigurations to prevent the original set modification
+        syncConfigurations = new HashSet<>(syncConfigurations);
+        syncConfigurations.add(syncConfiguration);
+        updateConfiguration(syncConfigurations, topics);
+    }
+
+    /**
+     * Removes shadow from sync configuration.
+     *
+     * @param syncConfigurations current configuration shadows
+     * @param topics             configuration of the component
+     * @param thingName          name of the thing
+     * @param shadowName         name of the shadow
+     *
+     * @throws RuntimeException if the handler is not configured with the core thing name
+     */
+    public void removeShadow(
+            Set<ThingShadowSyncConfiguration> syncConfigurations,
+            Topics topics,
+            String thingName,
+            String shadowName
+    ) {
+        validateHandlerIsConfigured();
+        ThingShadowSyncConfiguration syncConfiguration = validateThingAndShadowNames(thingName, shadowName);
+
+        logger.atInfo()
+                .kv(LOG_THING_NAME_KEY, syncConfiguration.getThingName())
+                .kv(LOG_SHADOW_NAME_KEY, syncConfiguration.getShadowName())
+                .log("removing thing shadow from sync on interaction");
+
+        if (!syncConfigurations.contains(syncConfiguration)) {
+            return;
+        }
+
+        if (thingIsNotAddedOnInteraction(syncConfigurations, thingName)) {
+            logger.atWarn()
+                    .kv(LOG_THING_NAME_KEY, syncConfiguration.getThingName())
+                    .kv(LOG_SHADOW_NAME_KEY, syncConfiguration.getShadowName())
+                    .log("can not remove thing shadow on interaction because the thing has other non-interaction "
+                            + "shadows");
+            return;
+        }
+
+        // clone syncConfigurations to prevent the set modification
+        syncConfigurations = new HashSet<>(syncConfigurations);
+        syncConfigurations.remove(syncConfiguration);
+        updateConfiguration(syncConfigurations, topics);
+    }
+
+    private void updateConfiguration(Set<ThingShadowSyncConfiguration> syncConfigurations, Topics topics) {
```

Comment:
Closing this thread (feel free to reopen) because we discussed it and agreed on storing the data in configuration so the users are able to reset the shadows added on interaction from IoT Core using deployment and configuration update (reset paths).